### DISCUSSION TITLE
How to filter inner collections to fetch only records with active status through query?

### DISCUSSION THREAD

**Discussion by arjundnair-fingent** [ChilliCream/hotchocolate#4254] [Q&A]
I have two objects, `A` and `Detail` as follows:

``` 
  public class A
  {
     public int Id {get; set;}
     public string Name {get; set;}
     public ICollection<Detail> Details {get; set;}
  }

  public class Detail
  {
     public int Id {get; set;}
     public string Name {get; set;}
     public int Status {get; set;}
  }

```

I need to write a GraphQL query to fetch the details of `A`. I need to apply a filter to the `Details` collection in `A` such that the result object will only have `Detail` data with `Status` **1**. If there is no record with status **1**, it should return an empty `Details` collection.

The query I tried was:

```
query(where: {details: {all: {status: {eq: 1}}}})
{
  name,
  details{
    name
  }
}
```

But this doesn't seem to work. It is not applying any filter and is returning all records. Where am I going wrong? Please help me.

**Comment by ghost** [comment]
You do not show which "query" you are trying to execute. If we assume you have implemented, for example, "getAllAs" as a query, then the graphql would be:

```
query {
    getAllAs {
        name,
        details(where: { status: { eq: 1 }}) {
            name
        }
    }
}
```

This also assumes you have filtering on the nested resolver for the details field.

**Reply by durajetz** [reply]
When I am trying to implement "where" clause inside my collection list it shows error, doesn't recognize the where filtering.
Is this to a specific version of HotChocolate (I am currently using version 11.3.8) ?

**Reply by PascalSenn** [reply]
@durajetz  @celerum-deryck  is correct. but you have to define [UseFiltering] on the 'details' property too. if you also want to project it to the database you need '[UseProjection]` on `getAllAs`
